wandb/wandb_run.py

```diff
@@ -661,6 +661,41 @@ def _add_viz(self, key, viz_id):
         }
         self.config.persist()
 
+    # Stores a singleton item to wandb config.
+    #
+    # A singleton in this context is a piece of data that is continually
+    # logged with the same value in each history step, but representing
+    # as a single item in the config.
+    #
+    # We do this to avoid filling up history with a lot of repeated uneccessary data
+    #
+    # Add singleton can be called be called many times in one run
+    # and it will only be updated when the value changes. The last value
+    # logged will be the one persisted to the server
+    def _add_singleton(self, type, key, value):
+        # Wrap te value with information
+        value_extra = {
+            'type': type,
+            'key': key,
+            'value': value
+        }
+
+        if not 'singletons' in self.config['_wandb']:
+            self.config._set_wandb('singletons', {})
```

Comment:
Do we need this special singletons area? We could store key directly under _wandb, and then check if the value is equal with a deep compare.

Reply:
No. I removed it.